### PR TITLE
openblas: Build a bottle for Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
```
gfortran -O2 -Wall -m64 -fopenmp -fPIC -march=skylake-avx512 -c -o sgeqrf.o sgeqrf.f
f951: Error: bad value (skylake-avx512) for -march= switch
Makefile:602: recipe for target 'sbdsvdx.o' failed
make[2]: *** [sbdsvdx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

**Comment by sjackman** [comment]
GCC 5.4.0 doesn't seem to support `-march=skylake-avx512`. CircleCI has evidently upgraded their bottling systems to Skylake. Any thoughts on how to resolve this issue? I have a Skylake machine at work, and I can reproduce the issue there. It also does not occur on non-Skylake machines.

**Comment by iMichka** [comment]
I had no time to look at this more in detail. I just saw there there was some back and forth upstream regarding these flags. I think it is an upstream bug.

**Comment by sjackman** [comment]
I agree that it's an upstream bug. Perhaps they'll have a patch for us if we're lucky.

**Comment by iMichka** [comment]
I pushed a quick fix, let's see if that helps.

File: Formula/openblas.rb [hunk 1]
```diff
@@ -21,14 +21,21 @@ class Openblas < Formula
...
+    depends_on "gcc@6"
+  end
 
   fails_with :clang if build.with? "openmp"
 
   def install
     ENV["DYNAMIC_ARCH"] = "1" if build.bottle?
     ENV["USE_OPENMP"] = "1" if build.with? "openmp"
 
+    inreplace "cmake/system.cmake", 'set (KERNEL_DEFINITIONS "${KERNEL_DEFINITIONS} -march=skylake-avx512")', 'set (KERNEL_DEFINITIONS "${KERNEL_DEFINITIONS}")' unless OS.mac?
```
Comment (sjackman) [inline Formula/openblas.rb]: Did this workaround work? And if so, can we remove the dependency on `gcc@6`?

**Comment by sjackman** [comment]
Could you please report this bug upstream if you have a chance?

One possible short-term workaround is that I can build a bottle at work in Docker on a non-AVX-512 machine.

**Comment by iMichka** [comment]
I opened an upstream issue while I continue to dig through the source code. https://github.com/xianyi/OpenBLAS/issues/1909

File: Formula/openblas.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def install
     ENV["USE_OPENMP"] = "1" if build.with? "openmp"
 
     # Must call in two steps
-    system "make", "CC=#{ENV.cc}", "FC=gfortran", "libs", "netlib", "shared"
+    system "make", "CC=#{ENV.cc}", "FC=gfortran", "libs", "netlib", "shared", *("NO_AVX512=1" unless OS.mac?)
```
Comment (sjackman) [inline Formula/openblas.rb]: Looks good to me if this fix works. Thanks, Michka!

**Comment by iMichka** [comment]
Ok it worked. Once #10629 is merged, I will pull this bottle.

**Comment by sjackman** [comment]
Thanks for this fix, Michka.
We need to remember to revert this PR when we switch to GCC ≥ 6.

**Pull request closed without merging**